### PR TITLE
Nimble: Added Walkthrough tutorial for phy_prph example. (IDFGH-9693)

### PR DESCRIPTION
**Pull request by ESPAbhinav**

### CODE REVIEW

File: examples/bluetooth/nimble/ble_phy/phy_prph/tutorial/Ble_Phy_Peripheral_Example_Walkthrough.md [hunk 1]
```diff
@@ -0,0 +1,412 @@
+# BLE Peripheral PHY Example Walkthrough
+
+## Introduction
+
+In this tutorial, the ble_phy peripheral example code for the espressif chipsets with BLE5.0 support is reviewed. This example aims at understanding how to establish connections on preferred PHY and changing LE PHY once the connection is established. The code implements a BLE Peripheral PHY, which establishes a connection on LE 1M PHY and switches to LE 2M PHY/CODED once the connection is established.
```
Comment (rahult-github) [inline examples/bluetooth/nimble/ble_phy/phy_prph/tutorial/Ble_Phy_Peripheral_Example_Walkthrough.md:5]: "This example aims at understanding how to establish connections on preferred PHY and changing LE PHY once the connection is established"

To 

"This example aims at understanding how to accept incoming connections on preferred PHY and process the request to change LE PHY once the connection is established"

File: examples/bluetooth/nimble/ble_phy/phy_prph/tutorial/Ble_Phy_Peripheral_Example_Walkthrough.md [hunk 2]
```diff
@@ -0,0 +1,412 @@
...
+    ble_hs_cfg.sm_sc = 0;
+#endif
+#ifdef CONFIG_EXAMPLE_BONDING
+    ble_hs_cfg.sm_our_key_dist = 1;
+    ble_hs_cfg.sm_their_key_dist = 1;
+#endif
+
+ ```
+
+The main function calls `ble_svc_gap_device_name_set()` to set the default device name. 'blecent_phy' is passed as the default device name to this function.
```
Comment (rahult-github) [inline examples/bluetooth/nimble/ble_phy/phy_prph/tutorial/Ble_Phy_Peripheral_Example_Walkthrough.md:204]: change "blecent_phy"  to "bleprph-phy"

File: examples/bluetooth/nimble/ble_phy/phy_prph/tutorial/Ble_Phy_Peripheral_Example_Walkthrough.md [hunk 3]
```diff
@@ -0,0 +1,412 @@
...
+    rc = ble_gap_adv_start(own_addr_type, NULL, BLE_HS_FOREVER,
+                           &adv_params, bleprph_gap_event, NULL);
+    if (rc != 0) {
+        MODLOG_DFLT(ERROR, "error enabling advertisement; rc=%d\n", rc);
+        return;
+    }
+}
+```
+
+To support increased messaging, BLE5.0 is added with extended advertising PDUs. These PDUs are broadcast-only events (like electronic beacons). The following function is used to make extended advertisements. In Bluetooth® 4.0, all advertising was done on 3 of the 40 – 2.4GHz ISM band channels. With Bluetooth 5 there are now two sets of advertising channels: primary and secondary. The primary advertising channels are the original 3 of the 40 advertising channels defined in Bluetooth 4. While the secondary advertising channels use the 37 fixed channels previously reserved for data.
```
Comment (rahult-github) [inline examples/bluetooth/nimble/ble_phy/phy_prph/tutorial/Ble_Phy_Peripheral_Example_Walkthrough.md:350]: Bluetooth 4 --> Bluetooth 4.0

File: examples/bluetooth/nimble/ble_phy/phy_prph/tutorial/Ble_Phy_Peripheral_Example_Walkthrough.md [hunk 4]
```diff
@@ -0,0 +1,412 @@
...
+
+1. `bleprph_advertise`
+2. `ext_bleprph_advertise`
+
+
+The following code represent `bleprph_advertise` function that sets the advertisement data which includes flags, advertising tx power,  device name, and service UUIDS.
+
+```c
+
+static void
```
Comment (rahult-github) [inline examples/bluetooth/nimble/ble_phy/phy_prph/tutorial/Ble_Phy_Peripheral_Example_Walkthrough.md:294]: This section can be removed. Just keep ext_bleprph_advertise.

File: examples/bluetooth/nimble/ble_phy/phy_prph/tutorial/Ble_Phy_Peripheral_Example_Walkthrough.md [hunk 5]
```diff
@@ -0,0 +1,412 @@
...
+    if (rc != 0) {
+        MODLOG_DFLT(ERROR, "error enabling advertisement; rc=%d\n", rc);
+        return;
+    }
+}
+```
+
+To support increased messaging, BLE5.0 is added with extended advertising PDUs. These PDUs are broadcast-only events (like electronic beacons). The following function is used to make extended advertisements. In Bluetooth® 4.0, all advertising was done on 3 of the 40 – 2.4GHz ISM band channels. With Bluetooth 5 there are now two sets of advertising channels: primary and secondary. The primary advertising channels are the original 3 of the 40 advertising channels defined in Bluetooth 4. While the secondary advertising channels use the 37 fixed channels previously reserved for data.
+
+
```
Comment (rahult-github) [inline examples/bluetooth/nimble/ble_phy/phy_prph/tutorial/Ble_Phy_Peripheral_Example_Walkthrough.md:352]: Add a line describing, that as s_current_phy value keeps changing , the advertising data also gets modified.

**Review by rahult-github — CHANGES_REQUESTED** [review]

**Comment by rahult-github** [comment]
sha=55fd7d43c1995ff2351da51aaaa4f2df6f71de2a

**Comment by rahult-github** [comment]
sha=f9a895c9a80a83be0e17557b77b1403b47021bd7

**Comment by Alvin1Zhang** [comment]
Thanks for contribution again, sorry for very slow turnaround, changes have been merged with https://github.com/espressif/esp-idf/commit/6e1d60ea4830f4eb7bbb2bb01541cf1d45443931.

**Pull request closed without merging**